c/misra/src/rules/RULE-16-4/EverySwitchShallHaveDefaultLabel.ql

```diff
@@ -0,0 +1,32 @@
+/**
+ * @id c/misra/every-switch-shall-have-default-label
+ * @name RULE-16-4: Every switch statement shall have a default label
+ * @description The requirement for a default label is defensive programming.
+ * @kind problem
+ * @precision very-high
+ * @problem.severity error
+ * @tags external/misra/id/rule-16-4
+ *       maintainability
+ *       readability
+ *       external/misra/obligation/required
+ */
+
+import cpp
+import codingstandards.c.misra
+
+from SwitchStmt switch, string message
+where
+  not isExcluded(switch, Statements1Package::everySwitchShallHaveDefaultLabelQuery()) and
+  not switch.hasDefaultCase() and
+  message = "has missing default clause."
+  or
+  exists(SwitchCase case, BreakStmt break |
+    switch.getDefaultCase() = case and
+    case.getFollowingStmt() = break and
```

Comment:
A bit of a silly corner case that is covered by this.

```cpp
  switch (p1) { // NON_COMPLIANT
  case 1:
    i++;
    break;
  default: {
    break;
  }
  }
```

Reply:
Addressed.